### PR TITLE
Feature books sevice

### PR DESCRIPTION
**Pull request by hugocasser**

### CODE REVIEW

File: reader/src/backend/BooksService/Core/Application/Services/AuthorsService.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using Application.Abstractions.Services;
+using Application.Dtos.Requests;
+using Application.Dtos.Requests.Authors;
+using Application.Dtos.Views.Authors;
+using Application.Exceptions;
+using Domain.Models;
+
+namespace Application.Services;
+
+public class AuthorsService(IAuthorsRepository authorsRepository): IAuthorsService
```
[the remark this answers is not in the document]
Comment (hugocasser) [reply reader/src/backend/BooksService/Core/Application/Services/AuthorsService.cs:11]: but the primary constructor is not fields, it's just parameters for constructors or methods.

[the remark this answers is not in the document]
Comment (Elizaveta564) [reply reader/src/backend/BooksService/Core/Application/Services/AuthorsService.cs]: They are simply called as private fields, that is, with an underscore, but they are not private

File: reader/src/backend/BooksService/Core/Application/Services/AuthorsService.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        var author = new Author
+        {
+            Id = Guid.NewGuid(),
+            FirstName = requestDto.FirstName,
+            LastName = requestDto.LastName,
+            BirthDate = requestDto.BirthDate,
+            DeathDate = requestDto.DeathDate,
+            Biography = requestDto.Biography
+        };
```
Comment (Elizaveta564) [inline reader/src/backend/BooksService/Core/Application/Services/AuthorsService.cs:25]: You need to use the mapper to get your author from the requestDto

File: reader/src/backend/BooksService/External/Infrastructure/Persistence/MongoDbContext.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    public IMongoCollection<Category> CategoriesCollection { get; set; }
+
+    public MongoDbContext(IOptions<MongoOptions> options, IMongoClient client)
+    {
+        var database = client.GetDatabase(options.Value.DatabaseName);
+        BooksCollection = database.GetCollection<Book>(options.Value.CollectionsNames.First());
+        AuthorsCollection = database.GetCollection<Author>(options.Value.CollectionsNames.ElementAt(1));
+        CategoriesCollection = database.GetCollection<Category>(options.Value.CollectionsNames.ElementAt(2));
+    }
+
```
Comment (Elizaveta564) [inline reader/src/backend/BooksService/External/Infrastructure/Persistence/MongoDbContext.cs:21]: extra line

**Review by Elizaveta564 — APPROVED** [review]

**Pull request merged**